pwndbg/commands/context.py

```diff
@@ -150,6 +152,87 @@ def contextoutput(section, path, clearing, banner="both", width=None):
+# This is not (sorry couldn't help it) "the yellow of the egg"
+# Using radare2 is suboptimal and will only be an intermediate step to have
+# ghidra decompiler within pwndbg.
+# But having ghidra in pwndgb is a lot more work as it requires quite some code and c/c++
+radare2 = {}
+
+config_context_ghidra = pwndbg.config.Parameter('context-ghidra',
+                                                  'always never if-no-source',
+                                                  'configures if or/when to try to decompile with ghidra (slow and requires radare2/r2pipe')
+def init_radare2(filename):
+    r2 = radare2.get(filename)
+    if r2:
+        return r2
+    import r2pipe
+    r2 = r2pipe.open(filename)
+    radare2[filename] = r2
+    r2.cmd("aaaa")
+    return r2
+
+parser = argparse.ArgumentParser()
+parser.description = """Show current function decompiled by ghidra"""
+parser.add_argument("func", type=str, default=None, nargs="?",
+                    help="Function to be shown. Defaults to current")
+@pwndbg.commands.ArgparsedCommand(parser, aliases=['ctx-ghidra'])
+def contextghidra(func):
+    print("\n".join(context_ghidra(func, with_banner=False, force_show=True)))
+
+def context_ghidra(func=None, target=sys.stdout, with_banner=True, width=None, force_show=False):
+    banner = [pwndbg.ui.banner("ghidra decompile", target=target, width=width)]
+    if config_context_ghidra == "never" and not force_show: return []
+    if config_context_ghidra == "if-no-source" and not force_show:
+        try:
+            f = open(gdb.selected_frame().find_sal().symtab.fullname())
+            f.close()
+        except:         # a lot can go wrong in search of source code.
+            return []   # we don't care what, just that it did not work out well...
+    filename = gdb.current_progspace().filename
+    try:
+        r2 = init_radare2(filename)
+        if not "pdg" in r2.cmd("LD").split("\n"):
+            return banner + ["radare2 plugin r2ghidra-dec must be installed and available from r2"]
+    except ImportError: # no r2pipe present
+        return banner + ["r2pipe not available, but required for r2->ghidra-bridge"]
+    if func == None:
+        try:
+            func = hex(pwndbg.regs[pwndbg.regs.current.pc])
+        except:
+            func = "main"
+    src = r2.cmdj("pdgj @"+func)
+    source = src.get("code", "")
+    curline = None
+    try:
+        cur = pwndbg.regs[pwndbg.regs.current.pc]
+        closest = 0
+        for off in [a.get("offset", 0) for a in src.get("annotations", [])]:
+            if abs(cur - closest) > abs(cur - off):
+                closest = off
+        pos_annotations = sorted([a for a in src.get("annotations", []) if a.get("offset") == closest],
+                         key=lambda a: a["start"])
+        if pos_annotations:
+            curline = source.count("\n", 0, pos_annotations[0]["start"])
+    except AttributeError: pass # If not running there is no current.pc
```

Comment:
Let's move `pass` to the next line

Reply:
Also what fires off `AttributeError`? We have quite broad `try ... except` here - it might be good to narrow it down to the specific need, otherwise, if some changes are made in the future and `AttributeError` fires off, we might not notice it.

Btw if it is for `pwndbg.regs.current.pc` to prevent running the command when we don't debug any target, then we should use `@pwndbg.commands.OnlyWhenRunning` decorator instead e.g. like here: https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/commands/stack.py#L16-L17

Reply:
> Also what fires off `AttributeError`? We have quite broad `try ... except` here - it might be good to narrow it down to the specific need, otherwise, if some changes are made in the future and `AttributeError` fires off, we might not notice it.
> 
> Btw if it is for `pwndbg.regs.current.pc` to prevent running the command when we don't debug any target, then we should use `@pwndbg.commands.OnlyWhenRunning` decorator instead e.g. like here: https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/commands/stack.py#L16-L17

It is for `pwndbg.regs.current.pc`, but `@pwndbg.commands.OnlyWhenRunning` is to strong.
The command can run just fine when not running, it just can not get the current line (obviously).
I'll narrow down the try to the a single line and surround the rest with an if.